src/components/screens/Cadastros/EditaisContratos/ContratosRelacionados.jsx

```diff
@@ -17,21 +18,96 @@ class ContratosRelacionados extends Component {
     super(props);
     this.state = {
       lotesSelecionados: [],
+      lotesNomesSelecionados: [],
+
       diretoriasSelecionadas: [],
-      empresasSelecionadas: []
+      diretoriasNomesSelecionadas: [],
+
+      empresasSelecionadas: [],
+      empresasNomesSelecionados: [],
+
+      contratos_datas: [
+        {
+          numero_contrato: null,
+          data_inicio: null,
+          data_fim: null
+        }
+      ],
+
+      formVigenciaContratos: ["vigenciaContrato0"]
     };
   }
 
-  atualizarDiretoriasSelecionadas(diretoriasSelecionadas) {
-    this.setState({ diretoriasSelecionadas });
+  handleField(field, value, key, indice) {
+    let contratos_datas = this.state.contratos_datas;
+    contratos_datas[key][field] = value;
+    this.setState({
+      ...this.state,
+      contratos_datas: contratos_datas
```

Comment:
quando a variável tem o mesmo nome não precisa colocar dois pontos
this.setState({ contratos_datas });

Reply:
ok